source/dub/packagemanager.d

```diff
@@ -377,21 +378,41 @@ class PackageManager {
 		Params:
 			path = NativePath to the root directory of the package
 			recipe_path = Optional path to the recipe file of the package
-			allow_sub_packages = Also return a sub package if it resides in the given folder
+			allow_sub_packages = Also return an already-loaded sub package if it resides in the given folder
 			mode = Whether to issue errors, warning, or ignore unknown keys in dub.json
+			name = Optional (sub-)package name if known in advance
 
 		Returns: The packages loaded from the given path
 		Throws: Throws an exception if no package can be loaded
 	*/
 	Package getOrLoadPackage(NativePath path, NativePath recipe_path = NativePath.init,
-		bool allow_sub_packages = false, StrictMode mode = StrictMode.Ignore)
+		bool allow_sub_packages = false, StrictMode mode = StrictMode.Ignore,
+		PackageName name = PackageName.init)
 	{
 		path.endsWithSlash = true;
-		foreach (p; this.m_internal.fromPath)
-			if (p.path == path && (!p.parentPackage || (allow_sub_packages && p.parentPackage.path != p.path)))
+		const nameString = name.toString();
+
+		foreach (p; this.m_internal.fromPath) {
+			if (p.path != path) continue;
+			if (!nameString.empty && p.name != nameString) continue;
+			if (!p.parentPackage || (allow_sub_packages && p.parentPackage.path != p.path))
```

Comment:
Hmm wait, this would reject an already loaded subpackage if it's defined in the parent recipe.